x-pack/plugins/maps/public/classes/joins/inner_join.ts

```diff
@@ -33,23 +40,25 @@ export function createJoinSource(
     return;
   }
 
-  if (
-    descriptor.type === SOURCE_TYPES.ES_TERM_SOURCE &&
-    descriptor.indexPatternId !== undefined &&
-    descriptor.term !== undefined
-  ) {
+  if (descriptor.type === SOURCE_TYPES.ES_DISTANCE_SOURCE && isSpatialSourceComplete(descriptor)) {
```

Comment:
In this PR should we also update the [LayerStatsCollector](https://github.com/nreese/kibana/blob/distance_source/x-pack/plugins/maps/common/telemetry/layer_stats_collector.ts#L150-L155) to capture the join type accurately?

Reply:
Thanks for pointing telemetry out. I have updated LayerStatsCollector to collect distance join telemetry